src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -150,6 +131,9 @@
   ([]
    {:where [:and
             [:= :active true]
+            [:or
+             [:not= :special_type "type/PK"]
```

Comment:
since types are hierarchical we should really be checking that the `special_type` isn't a PK or any type derived from PK (we don't have any now, but we might in the future). Basically we want to do 

```sql
special_type NOT IN ('type/PK', ...)
```

We can introspect the descendants of `type/PK` to construct the list, and luckily I already wrote a function to do just that to generate the HoneySQL form called `metabase.db/isa`. So you would want the HoneySQL here would be something like 

```clojure
[:or
 [:not (mdb/isa :special_type :type/PK)]
 ...]
```